File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/MahasiswaAnalogi.java
package id.ac.uph.pbo.pertemuan01;

public class MahasiswaAnalogi {

    // ── State ────────────────────────────────────────────────────────────────
    private String nama;
    private String nim;
    private double ipk;
    private int    semester;

    // ── Constructor ──────────────────────────────────────────────────────────
    public MahasiswaAnalogi(String nama, String nim, double ipk, int semester) {
        this.nama     = nama;
        this.nim      = nim;
        this.ipk      = ipk;
        this.semester = semester;
    }

    // ── Behavior ─────────────────────────────────────────────────────────────
    public void belajar() {
        System.out.println("[" + nama + "] sedang belajar…");
    }

    public void mengerjakanTugas(String namaTugas) {
        System.out.println("[" + nama + "] mengerjakan tugas: " + namaTugas);
    }

    public void hadirKuliah(String mataKuliah) {
        System.out.println("[" + nama + "] hadir di: " + mataKuliah);
    }

    public void ujian(String mataKuliah) {
        System.out.println("[" + nama + "] mengikuti ujian: " + mataKuliah);
    }

    // ── Tampilkan state ──────────────────────────────────────────────────────
    public void tampilkanState() {
        System.out.println("  Nama     : " + nama);
        System.out.println("  NIM      : " + nim);
        System.out.println("  IPK      : " + ipk);
        System.out.println("  Semester : " + semester);
    }
}

File: PYTHON/Pertemuan 01/MahasiswaAnalogi.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class MahasiswaAnalogi:
    def __init__(self, nama: str, nim: str, ipk: float, semester: int):
        self.nama = nama
        self.nim = nim
        self.ipk = ipk
        self.semester = semester

    def belajar(self):
        print(f"[{self.nama}] sedang belajar…")

    def mengerjakan_tugas(self, nama_tugas: str):
        print(f"[{self.nama}] mengerjakan tugas: {nama_tugas}")

    def hadir_kuliah(self, mata_kuliah: str):
        print(f"[{self.nama}] hadir di: {mata_kuliah}")

    def ujian(self, mata_kuliah: str):
        print(f"[{self.nama}] mengikuti ujian: {mata_kuliah}")

    def tampilkan_state(self):
        print(f"  Nama     : {self.nama}")
        print(f"  NIM      : {self.nim}")
        print(f"  IPK      : {self.ipk}")
        print(f"  Semester : {self.semester}")

